src/core/lib/security/credentials/google_default/google_default_credentials.cc

```diff
@@ -49,9 +49,11 @@
 
 /* -- Default credentials. -- */
 
-static int g_compute_engine_detection_done = 0;
-static int g_need_compute_engine_creds = 0;
+static int g_metadata_server_detection_done = 0;
+static int g_metadata_server_available = 0;
+static int g_is_on_gce = 0;
 static gpr_mu g_state_mu;
+static gpr_mu* g_polling_mu;
```

Comment:
Could you please document what the mutex is protecting? IIUC, it's the pollset and is_done, but I'm not entirely sure.

Reply:
Yes, you are right. It is protecting `pollent` and `is_done` bit. I added a comment to make it more clear.